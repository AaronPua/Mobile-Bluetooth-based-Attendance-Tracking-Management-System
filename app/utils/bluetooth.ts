import BLEAdvertiser from 'react-native-ble-advertiser';

// Uses the Apple code to pick up iPhones
const APPLE_ID = 0x4C;
const MANUF_DATA = [1,0];

export const startAdvertising = (uuidString: string, beaconName: string) => {
    BLEAdvertiser.setCompanyId(APPLE_ID);

    const options = {
        advertiseMode: 1, // ADVERTISE_MODE_BALANCED
        txPowerLevel: 1, // ADVERTISE_TX_POWER_LOW
        includeDeviceName: true,
        connectable: false,
        beaconName: beaconName
    }

    BLEAdvertiser.broadcast(uuidString, MANUF_DATA, options)
    .then(success => console.log(uuidString, "Adv Successful", success))
    .catch(error => console.log(uuidString, "Adv Error", error));
}

export const stopAdvertising = () => {
    BLEAdvertiser.stopBroadcast()
    .then(success => console.log("Stop Broadcast Successful", success))
    .catch(error => console.log("Stop Broadcast Error", error));
}

export const startScan = () => {
    
}