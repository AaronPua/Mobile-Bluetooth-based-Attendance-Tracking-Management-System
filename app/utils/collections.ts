import { Mongo } from '@meteorrn/core';

export const CoursesCollection = new Mongo.Collection("courses");
export const UsersCollection = new Mongo.Collection("users");
export const BeaconsCollection = new Mongo.Collection("beacons");
export const LessonsCollection = new Mongo.Collection("lessons");