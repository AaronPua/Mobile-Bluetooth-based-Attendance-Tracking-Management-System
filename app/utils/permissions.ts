// eslint-disable-next-line react-native/split-platform-components
import { Alert, PermissionsAndroid } from 'react-native';
import BLEAdvertiser from 'react-native-ble-advertiser';

export const requestLocationBluetoothPermissions = async () => {
    try {
        const granted = await PermissionsAndroid.request(
            PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION,
                {
                    title: 'Location Permission',
                    message: 'This app needs location permission to operate correctly.',
                    buttonNeutral: 'Ask Me Later',
                    buttonNegative: 'Cancel',
                    buttonPositive: 'OK',
                },
            );
        if (granted === PermissionsAndroid.RESULTS.GRANTED) {
            console.log('[Permissions]', 'Location Permission granted');
        } else {
            console.log('[Permissions]', 'Location Permission denied');
        }

        const bluetoothActive = await BLEAdvertiser.getAdapterState().then(result => {
            console.log('[Bluetooth]', 'Bluetooth Status', result)
                return result === "STATE_ON";
            }).catch(error => { 
                console.log('[Bluetooth]', 'Bluetooth Not Enabled', error)
                return false;
            });

        if (!bluetoothActive) {
            Alert.alert(
                'Example requires bluetooth to be enabled',
                'Would you like to enable Bluetooth?',
                [
                    {
                        text: 'Yes',
                        onPress: () => BLEAdvertiser.enableAdapter(),
                    },
                    {
                        text: 'No',
                        onPress: () => console.log('Do Not Enable Bluetooth Pressed'),
                        style: 'cancel',
                    },
                ],
            )
        }
    } catch (err) {
        console.warn(err);
    }
}