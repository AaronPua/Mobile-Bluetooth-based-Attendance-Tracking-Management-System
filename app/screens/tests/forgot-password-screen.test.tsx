import React from "react"
import { fireEvent, waitFor } from '@testing-library/react-native'
import { renderWithNativeBase } from './utils/jest-setup'
import { ForgotPasswordScreen } from '../index'
import { Accounts } from '@meteorrn/core'

jest.mock('@react-navigation/native', () => {
    return {
        createNavigatorFactory: jest.fn(),
        useNavigation: jest.fn(),
        createNavigationContainerRef: jest.fn(),
    }
});

jest.mock('react-native-vector-icons', () => 'Icon');

describe('<ForgotPasswordScreen />', () => {

    it('renders correctly', () => {
        const tree = renderWithNativeBase(<ForgotPasswordScreen navigation={undefined} route={undefined} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('forgot password method gets called', async () => {
        const spyMethod = jest.spyOn(Accounts, 'forgotPassword');
        const { getByLabelText } = renderWithNativeBase(<ForgotPasswordScreen navigation={undefined} route={undefined} />);

        const emailInput = getByLabelText(/Email/i);
        const sendButton = getByLabelText(/Send/i);

        fireEvent.changeText(emailInput, 'not_exist@test.com');
        fireEvent.press(sendButton);

        await waitFor(() => {
            expect(sendButton).toBeDisabled();
            expect(spyMethod).toHaveBeenCalled();
        });

        jest.clearAllMocks();
    });
});