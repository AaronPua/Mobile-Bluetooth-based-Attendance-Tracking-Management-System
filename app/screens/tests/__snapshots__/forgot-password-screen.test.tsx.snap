// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForgotPasswordScreen /> renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    _text={
      Object {
        "color": "darkText",
      }
    }
    contentContainerStyle={
      Array [
        Object {},
        Object {
          "dataSet": Object {},
        },
      ]
    }
    dataSet={Object {}}
    style={
      Object {
        "backgroundColor": "#f1f5f9",
        "flex": 1,
      }
    }
  >
    <View>
      <View
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": "#6200ee",
          }
        }
      />
      <View
        dataSet={Object {}}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#6200ee",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 12,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 12,
            "width": "100%",
          }
        }
      >
        <View
          dataSet={Object {}}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#FFFFFF",
                "fontFamily": undefined,
                "fontSize": 20,
                "fontStyle": "normal",
                "fontWeight": "700",
                "letterSpacing": 0,
                "lineHeight": 30,
                "textDecorationLine": undefined,
              }
            }
          >
            Password Reset
          </Text>
        </View>
      </View>
      <View
        dataSet={Object {}}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <View
          dataSet={Object {}}
          style={
            Object {
              "maxWidth": 290,
              "paddingBottom": 32,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 32,
              "width": "90%",
            }
          }
        >
          <Text
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#1f2937",
                "fontFamily": undefined,
                "fontSize": null,
                "fontStyle": "normal",
                "fontWeight": "600",
                "letterSpacing": 0,
                "lineHeight": 30,
                "textDecorationLine": undefined,
              }
            }
          >
            Forgot Password?
          </Text>
          <Text
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#4b5563",
                "fontFamily": undefined,
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "500",
                "letterSpacing": 0,
                "lineHeight": 17.5,
                "marginTop": 4,
                "textDecorationLine": undefined,
              }
            }
          >
            A password reset email will be sent
          </Text>
          <View
            dataSet={Object {}}
            style={
              Object {
                "marginTop": 20,
              }
            }
          >
            <View
              dataSet={Object {}}
              isInvalid={false}
              style={
                Object {
                  "marginBottom": 8,
                  "width": "100%",
                }
              }
            >
              <View
                dataSet={Object {}}
                feedbackId="field-react-aria-1-feedback"
                hasFeedbackText={false}
                hasHelpText={false}
                helpTextId="field-react-aria-1-helptext"
                isDisabled={false}
                isInvalid={false}
                isReadOnly={false}
                isRequired={false}
                labelId="field-react-aria-1-label"
                nativeID="field-react-aria-1-label"
                setHasFeedbackText={[Function]}
                setHasHelpText={[Function]}
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "marginBottom": 8,
                    "marginRight": 12,
                  }
                }
              >
                <Text
                  dataSet={Object {}}
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "color": "#1f2937",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "500",
                      "letterSpacing": 0,
                      "lineHeight": 21,
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  Email
                </Text>
              </View>
              <TextInput
                accessibilityLabel="Email Input"
                accessible={true}
                allowFontScaling={true}
                dataSet={Object {}}
                disabled={false}
                editable={true}
                isRequired={false}
                nativeID="field-react-aria-1"
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onKeyPress={[Function]}
                placeholderTextColor="#a3a3a3"
                readOnly={false}
                rejectResponderTermination={true}
                required={false}
                secureTextEntry={false}
                selectionColor="#1f2937"
                style={
                  Array [
                    Object {
                      "borderColor": "#e5e5e5",
                      "borderRadius": 4,
                      "borderWidth": 1,
                      "color": "#1f2937",
                      "fontFamily": undefined,
                      "fontSize": 12,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "overflow": "hidden",
                      "paddingBottom": 8,
                      "paddingLeft": 8,
                      "paddingRight": 8,
                      "paddingTop": 8,
                      "width": undefined,
                    },
                    Object {},
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View
              _text={
                Object {
                  "color": "lightText",
                }
              }
              accessibilityLabel="Send Button"
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              colorScheme="indigo"
              dataSet={Object {}}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onHoverIn={[Function]}
              onHoverOut={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#6366f1",
                  "borderRadius": 4,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginTop": 8,
                  "paddingBottom": 8,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 8,
                }
              }
            >
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  dataSet={Object {}}
                  style={Object {}}
                >
                  <Text
                    dataSet={Object {}}
                    style={
                      Object {
                        "backgroundColor": undefined,
                        "color": "#FFFFFF",
                        "fontFamily": undefined,
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "500",
                        "letterSpacing": 0,
                        "lineHeight": 21,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    Send
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaProvider>
`;
